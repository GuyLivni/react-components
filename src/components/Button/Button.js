import React from 'react';
import PropTypes from 'prop-types';

import PrimaryButton from './components/PrimaryButton';
import SecondaryButton from './components/SecondaryButton';

const Buttons = {
  primary: PrimaryButton,
  secondary: SecondaryButton,
};

const Button = ({ variant, ...props }) => {
  const ButtonVariant = Buttons[variant];
  return <ButtonVariant {...props} />;
};

Button.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary']),
};

Button.defaultProps = {
  variant: 'primary',
};

export default Button;
