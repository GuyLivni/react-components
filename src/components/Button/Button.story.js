import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs';

import Button from './Button';

const options = ['primary', 'secondary'];

storiesOf('Button', module).add('Default', () => (
  <Button variant={select('variant', options, options[0])}>Button</Button>
));
