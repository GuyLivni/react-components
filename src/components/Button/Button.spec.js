import React from 'react';
import Button from './Button';

describe('Button', () => {
  it('should render a primary button', () => {
    const component = create(<Button>Primary</Button>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render a secondary button', () => {
    const component = create(
      <Button variant="secondary">Secondary</Button>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Button>Primary</Button>);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
