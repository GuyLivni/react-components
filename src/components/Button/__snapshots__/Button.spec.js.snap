// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a primary button 1`] = `
.c0 {
  font-size: 14px;
  height: 50px;
  font-weight: 600;
  padding: 0 45px;
  text-rendering: geometricPrecision;
  line-height: inherit;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: top;
  cursor: pointer;
  background: #0074c8;
  color: #fff;
  border: 1px solid #0074c8;
  border-radius: 5px;
  -webkit-transition: background-color 0.2s ease,border-color 0.2s ease;
  transition: background-color 0.2s ease,border-color 0.2s ease;
}

.c0:hover,
.c0:focus {
  background-color: #0065ad;
  border-color: #0065ad;
}

<button
  className="c0"
>
  Primary
</button>
`;

exports[`Button should render a secondary button 1`] = `
.c0 {
  font-size: 14px;
  height: 50px;
  font-weight: 600;
  padding: 0 45px;
  text-rendering: geometricPrecision;
  line-height: inherit;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: top;
  cursor: pointer;
  background: #e0e1e2;
  color: #888787;
  border: 1px solid #e0e1e2;
  border-radius: 5px;
  -webkit-transition: background-color 0.2s ease,border-color 0.2s ease;
  transition: background-color 0.2s ease,border-color 0.2s ease;
}

.c0:hover,
.c0:focus {
  background-color: #cacbcd;
  border-color: #cacbcd;
}

<button
  className="c0"
>
  Secondary
</button>
`;
